### PR TITLE
Styling/hamburger menu

### PR DESCRIPTION
**Pull request by BartalD**
### Why?
- VFI wanted the hamburger menu on mobile and tablet to match the design from [Figma](https://www.figma.com/file/eOshLuKuFSfE7gWDYgVcFb/VFI-Tourism---mobile?node-id=0%3A736&mode=dev)

### What has been done?
- I've tried as best as I can to compartmentalize the different segments and use Twig to include them.
  - This was done by extracting the code to different smaller html files.
  - Likewise, the .scss code was extracted into seperate files.
  - This will allow us to use these segments elsewhere, and also makes the code easier to read and edit. Especially main.scss and globals.scss have become highly spaghettified.

#### Details
- Added .svg assets [96efc4265341226d3c6ee9677f8d5f450c3a276c]
- Added media queries from Sendistovan to the Variables file [ed4c3d9b1c12832ee8b2a4fd5f252897cab3ab5b]
  - The plan is that all files will use these dimensions
- Styled the What's On content from the Accordion menu in the side-menu [d8ac160]
- Extracted scss code from global.scss [f1ed2c5]
- Added side menu for mobile/tablet [8cb7ec1]
- Changed What's On accordion from grid to flex [ab1b280]
- Follow Us (Most likely won't be used) [10f4afc]
- Explore the map section [e3ccf55]
- Added 'chosenLanguage' [df96e8c]
- 
#### Objective
- To verify that the side-menu styling is consistent and functional across different screen sizes (desktop, tablet, and mobile) in the pull request
- To verify that it matches the syling from [Figma](https://www.figma.com/file/eOshLuKuFSfE7gWDYgVcFb/VFI-Tourism---mobile?node-id=0%3A736&mode=dev)

#### Preconditions
- The pull request with side-menu styling changes is ready for testing
- Make three custom viewports for your browser:
  - Desktop: 1280px and up
  - Tablets: 768px - 1279px
  - Mobile: up to 767px

#### Desktop Testing
- Access the application using a desktop browser.
- Verify that the side-menu styling is visually consistent with the design specifications.
- Check for any layout issues, such as alignment problems or overlapping elements.
- Test the functionality of the side menu, ensuring that it opens and closes smoothly.
- Verify that links and buttons within the side menu are clickable and lead to the correct destinations.

#### Tablet Testing
- Access the application using a tablet-sized viewport (With the dimensions from Sendistovan).
- Verify that the side-menu styling adapts to the tablet screen size gracefully.
- Test the functionality of the side menu, including open and close actions, as well as interactions with menu items.
- Check for any touch-related issues, such as unresponsiveness or overlapping elements.

#### Mobile Testing
- Access the application using a mobile-sized viewport (or use a mobile emulator).
- Verify that the side-menu styling is optimized for mobile screens.
- Test the touch interactions for opening and closing the side menu.
- Check for any issues related to responsiveness, such as text or button size, that may affect usability.
- Verify that the side menu and its content are visually appealing and functional on mobile devices.

### CODE REVIEW

**Comment by Horib** [comment]
Can you link to monday.com task?

**Review by Horib — CHANGES_REQUESTED** [review]
### Desktop
firefox_QTKFFJAmal


### Tablet & Mobile
firefox_RBp1bDo9SF
firefox_gnPnQ3Kk9B
firefox_06nPeMmGg6
firefox_p5PQWGnkts
firefox_CEvrfT3HSS

### What's On
firefox_WE1K5m07Vt

**Comment by BartalD** [comment]
# Review-points from Rasmus

### Desktop

- [ ] Menu does not open anymore on desktop
  

### Mobile & Tablet

- [ ] First letter capitalization on the language pane (Figma is not capitalized)
  
- [ ] Scroll overflow problem on menu (was not able to reproduce)
  
- [ ] Missing title for Events, Place etc. (was not able to reproduce)
  
- [ ] Spacing between links (Activities f.x.) (this has to be another pull request, since I need to atomize this large one)
  
- [ ] Missing appearances Image Modules (See & Do) (this has to be another pull request, since I need to atomize this larger one)
  

### All platforms

- [ ] Does not work through VFI

**Review by paturvita — CHANGES_REQUESTED** [review]
You probably need to delete the font directory and the needsharebutton scripts from the dist folder, these don't look like they belong

There might be a number of files accidentally added, which have been deleted in earlier commits from other people. I've commented on some of them, but might not have caught all.

DESKTOP:

- In the "see & do" menu, we are getting [object object] text on images with missing data. Missing data should be properly handled. This menu point will also link to ...[object object]
objectobject

- In the "plan your stay" menu, the currency pane has misaligned elements
plan your stay

- In the "plan your stay" menu, the good to know pane is echoing out a serialized string. This should be handled properly and maybe the root cause needs to be handled. It is possible that the string is double-serialized and then only unserialized once before being presented.
goodtoknow

TABLET:
- The "sites" menu item should be normal font weight
- The "explore the map" menu item should be centered. 
Screendump with both issues:
sites

Stopping review, the PR does not match Figma, and there has been some misunderstanding on what was expected in the review as well as what to implement

File: assets/Variables/Variables.scss
[the remark this answers is not in the document]
Comment (BartalD) [reply assets/Variables/Variables.scss:1]: That is true! But my intent was to give them unique names to distinguish them from any other media queries, such that I, in the future, will have an easier task of cleaning up in them.

File: views/menus/sideMenus/mobileTabletSideMenu/style/sideMenu.scss
[the remark this answers is not in the document]
Comment (BartalD) [reply views/menus/sideMenus/mobileTabletSideMenu/style/sideMenu.scss:1]: My intent was to give them unique names to distinguish them from any other media queries, such that I, in the future, will have an easier task of cleaning up in them. The 'VFI' prefix is used such that it is clear that it is the 'official' color-codes, media-queries etc. that VFI have given us from Figma and Invision

File: views/modules/social_buttons/social_buttons.html
[the remark this answers is not in the document]
Comment (BartalD) [reply views/modules/social_buttons/social_buttons.html:1]: Not my code. Note to self: Look into this

File: git-deploy/deploy.log
[the remark this answers is not in the document]
Comment (BartalD) [reply git-deploy/deploy.log:1]: [Rasmus](@Horib), do you know what these are for?

File: git-deploy/deploy.php
[the remark this answers is not in the document]
Comment (BartalD) [reply git-deploy/deploy.php:1]: [Rasmus](@Horib), do you know what these are for?

File: git-deploy/deployer.php
[the remark this answers is not in the document]
Comment (BartalD) [reply git-deploy/deployer.php:1]: [Rasmus](@Horib), do you know what these are for?

File: views/main/index.html
[the remark this answers is not in the document]
Comment (BartalD) [reply views/main/index.html:1]: I have no idea how I have ended up as authoring this code. I don't recognize this.
[Rasmus](@Horib), do you know how this has happened?

File: views/modules/social_buttons/social_buttons.html
[the remark this answers is not in the document]
Comment (BartalD) [reply views/modules/social_buttons/social_buttons.html:1]: I have no idea how I have ended up as authoring this code. I don't recognize this.
[Rasmus](@Horib), do you know how this has happened?

**Comment by BartalD** [comment]
# Pull-Review from [Pætur](@paturvita)

## Unwanted Files

- [ ] Delete font-directory and needsharebutton scripts from Dist-folder
  
- [ ] Delete accidentally added files from previous commits
  

## Desktop

- [ ] [object Object] in See & Do menu
  
  - This bug is live on preview. So it seems that it is a bug from previous commits?
    
- [ ] In the "plan your stay" menu, the currency pane has misaligned elements
  
  - This bug is live on preview. So it seems that it is a bug from previous commits?
    
- [ ] In the "plan your stay" menu, the good to know pane is echoing out a 
  serialized string. This should be handled properly and maybe the root 
  cause needs to be handled. It is possible that the string is 
  double-serialized and then only unserialized once before being 
  presented.
  
  - This bug is live on preview. So it seems that it is a bug from previous commits?
    

## Tablet

- [ ] The "sites" menu item should be normal font weight
  
- [ ] The "explore the map" menu item should be centered.
  
  - This had been fixed. Not sure how it ended up back in the code. But 'explore map' is currently out of scope, so I will delete it.

**Comment by BartalD** [comment]
[Pætur](@paturvita) and [Rasmus](@Horib), can you re-review this? Regarding the content of the dropdown menus (images and columnization of text), I'm not able to do this at the moment, because the layout of both text and images is user-defined. And therefore I cannot make any general rules for the layout at this moment.

**Comment by BartalD** [comment]
The submenus now have two columns for tablet. @paturvita and @Horib

**Review by paturvita — CHANGES_REQUESTED** [review]
Columns in menu are still not acting like they should. Rejecting early instead of going indepth in review

**Comment by paturvita** [comment]
- Language should not be in all-caps. Image:
Language
- Overflow does not include the "sites" menu item. To reproduce, expand some menu item so that the menu overflows its container and observe that "sites" disappears from the menu. screenshot:
overflow
- Menu items are not spaced as in Figma. Difficult to give perfect pixel values, but the difference is visually apparent.
line height
- Spacing between icon and text is not the same as in Figma. Screenshot
whatson menu
- Language list is not aligned as in Figma. Languages should be aligned to same vertical line as the menu point headers.
Languages
- In tablet, the whatson menu points are not shown in the same vertical ratio as Figma (portrait-like ratio) see img below
- In tablet, whatson menu points contain additional text not shown in figma. see img
whatsonimgs

**Comment by BartalD** [comment]
# Fixes from Pull-Review from [Pætur](@paturvita)
- [x] Language should not be in all-caps
  
- [x] Overflow does not include the "sites" menu item
  
- [x] Menu items are not spaced as in Figma
  
- [x] Spacing between icon and text is not the same as in Figma
  
- [x] Language list is not aligned as in Figma
  
- [x] In tablet, the whatson menu points are not shown in the same vertical ratio as Figma (portrait-like ratio)
  
- [x] In tablet, whatson menu points contain additional text not shown in figma

**Pull request merged**